source/error_report/src/main.rs

```diff
@@ -0,0 +1,221 @@
+
+    let mut versions = Map::new();
+    versions.insert("z3-version".to_string(), Value::String(z3_version));
+    versions.insert("verus-version".to_string(), Value::String(verus_version));
+
+    let mut time = Map::new();
+    time.insert("verus-time".to_string(), Value::Float(verus_time));
+
+    let mut output = Map::new();
+    output.insert("stdout".to_string(), Value::String(stdout));
+    output.insert("stderr".to_string(), Value::String(stderr));
+
+    let mut map = Map::new();
+    map.insert(
+        "title".to_string(),
+        Value::String("Error report file - details and dependencies".to_string()),
+    );
+    map.insert("report-schema-version".into(), Value::String("1".to_string()));
+    map.insert("command-line-arguments".into(), Value::Table(command_line_arguments));
+    map.insert("verus-time".into(), Value::Table(time));
+    map.insert("versions".into(), Value::Table(versions));
+    map.insert("verus-output".into(), Value::Table(output));
+    Value::Table(map)
+}
+
+/// Uses the user input file to find the .d file, parse the dependencies,
+/// and write each dependency to the zip file.
+///
+/// Parameters: file_path: a String representation of the path to the input file
+///
+/// Returns:    the names of the .d file and zip file for book-keeping purposes
+pub fn zip_setup(file_path: String) -> (String, String) {
+    let file_name_path = Path::new(&file_path);
+    let temp_file_name =
+        file_name_path.with_extension("d").file_name().unwrap().to_string_lossy().to_string();
+    let mut deps = get_dependencies(&temp_file_name);
+    deps.push("error_report.toml".to_string());
+    let zip_file_name = write_zip_archive(deps);
+    (temp_file_name, zip_file_name)
+}
+
+/// Turns the .d file that lists each of the input files' dependencies
+/// and turns them into a vector of Strings for easier data manipulation
+///
+/// Parameters: file_name: The name of the previously generated .d file
+///
+/// Returns:    a vector containing each dependency of the input file as an
+///             individual string
+fn get_dependencies(file_name: &String) -> Vec<String> {
+    let file = File::open(file_name).expect("Couldn't open file!");
+    let mut reader = BufReader::new(file);
+    let mut dependencies = String::new();
+    reader.read_line(&mut dependencies).expect("Could not read the first line");
+    dependencies.split_whitespace().skip(1).map(|x| x.to_string()).collect()
+}
+
+/// Creates a zip file from a given list of files to compress
+///
+/// Parameters: deps: A vector of strings representing files to be compressed
+///                    (in this context, each file is a dependency of the input)
+///
+/// Returns:    The name of the created zip file
+fn write_zip_archive(deps: Vec<String>) -> String {
+    let local: DateTime<Local> = Local::now();
+    let date = local.to_string();
+    let mut zip_file_name = date[0..19].to_string();
```

Comment:
Can we construct the format we want, instead of just truncating the output of `to_string` and replacing characters?